flare/gp_from_aimd.py

```diff
@@ -250,14 +252,27 @@ def run(self):
 
             if i < train_frame:
                 # Get max uncertainty atoms
-                std_in_bound, train_atoms = is_std_in_bound_per_species(
+                std_in_bound, std_train_atoms = is_std_in_bound_per_species(
                     rel_std_tolerance=self.rel_std_tolerance,
                     abs_std_tolerance=self.abs_std_tolerance,
                     noise=self.gp.hyps[-1], structure=cur_frame,
                     max_atoms_added=self.max_atoms_from_frame,
                     max_by_species=self.train_env_per_species)
 
-                if not std_in_bound:
+                # Get max force error atoms
+                force_in_bound, force_train_atoms = \
+                    is_force_in_bound_per_species(
+                        abs_force_tolerance=self.abs_force_tolerance,
+                        predicted_forces=cur_frame.forces,
+                        label_forces=dft_forces,
+                        structure=cur_frame,
+                        max_atoms_added=self.max_atoms_from_frame,
+                        max_by_species=self.train_env_per_species)
+
+                if (not std_in_bound) or (not force_in_bound):
```

Comment:
There is this functionality! If the user doesn't specify species limits, then the logic for std_in_bound_by_species will not limit it by species counts. If the user doesn't specify a relative / absolute std cutoff then it will be skipped; if the user doesn't specify a force cutoff then it'll be skipped too. :)  If you specify both, it uses the union of the two. If you set both to 0, then it will just run the GP without validating the errors.

Reply:
Oh, oh. I missed that. That's terrific!